app/models/manageiq/providers/openstack/base_metrics_capture.rb

```diff
@@ -94,22 +94,26 @@ def meter_names(metric_capture_module)
   end
 
   def find_meter_counters(metric_capture_module, resource_filter, metadata_filter, log_header)
-    counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
-
+    # cache counters to prevent too frequent calls to Gnocchi API
+    cache_with_timeout(:counters, 60.minutes) do
+      counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
+    end
```

Comment:
The `cache_with_timeout` method defines a class-level singleton so this should be pulled out of the method and moved to the class level.  Something like `cache_with_timeout(:counters_by_vm, 60.minutes) { Hash.new }` then you should use the ems.id and vm.ems_ref as keys into the hash so that one metrics capture worker across multiple EMSs won't have hash collisions.